src/rust/ide/lib/view/src/graph_editor.rs

```diff
@@ -1830,29 +1966,38 @@ fn new_graph_editor(app:&Application) -> GraphEditor {
     //   better updates from engine services (see https://github.com/enso-org/enso/issues/441 ).
     trace inputs.set_expression_type;
 
+    }
+
 
     // === Move Edges ===
+    frp::extend! { network
 
     detached_edge           <- any(&inputs.some_edge_targets_detached,&inputs.some_edge_sources_detached);
-    cursor_pos_on_detach    <- cursor_pos_in_scene.sample(&detached_edge);
-    edge_refresh_cursor_pos <- any (cursor_pos_on_detach,cursor_pos_in_scene);
+    update_edge             <- any(detached_edge,on_new_edge_source,on_new_edge_target);
+    cursor_pos_on_update    <- cursor_pos_in_scene.sample(&update_edge);
+    edge_refresh_cursor_pos <- any (cursor_pos_on_update,cursor_pos_in_scene);
 
-    is_hovering_output <- inputs.hover_node_output.map(|target| target.is_some());
+    is_hovering_output <- inputs.hover_node_output.map(|target| target.is_some()).sampler();
     hover_node         <- inputs.hover_node_output.unwrap();
 
-    edge_refresh_on_node_hover       <- all(edge_refresh_cursor_pos,hover_node);
-    edge_refresh_cursor_pos_no_hover <- edge_refresh_cursor_pos.gate_not(&is_hovering_output);
+    edge_refresh_on_node_hover        <- all(edge_refresh_cursor_pos,hover_node).gate(&is_hovering_output);
+    edge_refresh_cursor_pos_no_hover  <- edge_refresh_cursor_pos.gate_not(&is_hovering_output);
+    edge_refresh_cursor_pos_on_hover  <- edge_refresh_on_node_hover.map(|(pos,_)| *pos);
 
-    eval edge_refresh_cursor_pos ((position) {
-        edges.detached_target.for_each(|id| {
+    refresh_target      <- any(&edge_refresh_cursor_pos_on_hover,&edge_refresh_cursor_pos_no_hover);
+    refresh_source      <- edge_refresh_cursor_pos_no_hover.map(|pos| *pos);
```

Comment:
what does this line do? If im correct, this does not do anything.

Reply:
This is mostly for naming ourpooses to make clear how this is used next.